### PR TITLE
Express Server setup

### PR DESCRIPTION
**Pull request by Bl4ckH4wkGER**
Hi Lynne & Fletcher,

Please find my initial setup and file structure for your review in this PR. Some if it was inspired by past assignments though I adjusted necessary parameters, etc.

From here, we should be able to `npm install` and build and test via postman.

Please let me know if you see anything amiss. Happy to make any changes as necessary to provide the foundation we need.

Cheers,
Sven

### CODE REVIEW

File: modals/authentication.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const mongoose = require('mongoose');
```
Comment (ilynne) [inline modals/authentication.js:1]: The folder name has a typo -- it should be "models."

File: daos/authentication.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const mongoose = require('mongoose');
```
Comment (ilynne) [inline daos/authentication.js:1]: I'm not going to request changes on these, but we don't actually need any of these files yet. We just need the empty app. The DAOs should be added when that issue is being added.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "main": "index.js",
+  "scripts": {
+    "start": "nodemon -e js,json,mustache index.js",
+    "test": "jest --coverage --runInBand",
+    "tdd": "jest --watch --runInBand"
+  },
+  "dependencies": {
+    "express": "^4.17.1",
+    "mongoose": "^5.9.21",
+    "mustache-express": "^1.3.0"
```
Comment (ilynne) [inline package.json:14]: I don't think we actually need mustache for this project. Let's leave it out until we do.

File: routes/authentication.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+const { Router, query } = require("express");
+const jwt = require('jsonwebtoken');
```
Comment (ilynne) [inline routes/authentication.js:2]: We have not decided whether to use jwt for authentication yet. I'm on the fence -- I don't like the fact that the tokens can't be invalidated. So let's not require it for now.

Also, as I said above, we don't need any of these route files (except index.js) yet. We should add them as the issues are work on. But let's leave them in.

File: modals/authentication.js (hunk 1 above)
Comment (Bl4ckH4wkGER) [reply modals/authentication.js:1]: Fixed.

File: daos/authentication.js (hunk 2 above)
Comment (Bl4ckH4wkGER) [reply daos/authentication.js:1]: Sorry, I got a little ahead of myself there it seems.

File: package.json (hunk 3 above)
Comment (Bl4ckH4wkGER) [reply package.json:14]: Fixed.

File: routes/authentication.js (hunk 4 above)
Comment (Bl4ckH4wkGER) [reply routes/authentication.js:2]: Removed and sorry again, I got a little ahead of myself there it seems.

**Comment by Bl4ckH4wkGER** [comment]
@ilynne I pushed up the adjustments you had requested.

**Comment by Bl4ckH4wkGER** [comment]
I will go ahead and merge this at this point so we can continue to build the actual codebase. We can always delete or rename files as needed.

**Pull request merged**